vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java

```diff
@@ -143,6 +145,16 @@ public void deleteSpaceLink(String linkId) {
         spaceLinkRepo.delete((SpaceLink) link);        
     }
     
+    @Override
+    public void deleteSpaceLinkBySource(String sourceId) throws SpaceLinkDoesNotExistException {
```

Comment:
doesn't that possibly delete several space links? This should be reflected by a plural "SpaceLinks" in method name